features/checkout.feature

```diff
@@ -120,3 +120,104 @@ Feature: hub checkout <PULLREQ-URL>
     When I run `hub checkout https://github.com/mojombo/jekyll/pull/77 mycustombranch`
     Then "git fetch origin +refs/heads/fixes:refs/remotes/origin/fixes" should be run
     And "git checkout -b mycustombranch --track origin/fixes" should be run
+
+  Scenario: Checkout a pull request with --no-branch
+    Given the GitHub API server:
+      """
+      get('/repos/mojombo/jekyll/pulls/77') {
+        halt 406 unless request.env['HTTP_ACCEPT'] == 'application/vnd.github.v3+json;charset=utf-8'
+        json :head => {
+          :ref => "fixes",
+          :repo => {
+            :owner => { :login => "mislav" },
+            :name => "jekyll",
+            :private => false
+          }
+        }, :base => {
+          :repo => {
+            :name => 'jekyll',
+            :html_url => 'https://github.com/mojombo/jekyll',
+            :owner => { :login => "mojombo" },
+          }
+        }
+      }
+      """
+    When I run `hub checkout --no-branch https://github.com/mojombo/jekyll/pull/77 -q`
+    Then "git fetch origin pull/77/head" should be run
+    And "git checkout FETCH_HEAD -q" should be run
+
+  Scenario: Checkout a pull request with --no-branch and a branch name
+    Given the GitHub API server:
+      """
+      get('/repos/mojombo/jekyll/pulls/77') {
+        halt 406 unless request.env['HTTP_ACCEPT'] == 'application/vnd.github.v3+json;charset=utf-8'
```

Comment:
Actually, the whole `Given the GitHub API server` block should not be needed at all in this scenario, since the flag parsing should abort the command early and no API communication should take place. I think.